bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -502,45 +502,11 @@ impl Timeline {
     pub async fn send_reply(
         &self,
         msg: Arc<RoomMessageEventContentWithoutRelation>,
-        event_id: String,
+        reply_params: ReplyParameters,
     ) -> Result<(), ClientError> {
-        let event_id = EventId::parse(event_id)?;
-        self.inner
-            .send_reply((*msg).clone(), event_id, EnforceThread::MaybeThreaded)
-            .await
-            .map_err(|err| anyhow::anyhow!(err))?;
-        Ok(())
-    }
-
-    /// Send a message on a thread.
-    ///
-    /// If the replied to event does not have a thread relation, it becomes the
-    /// root of a new thread.
-    ///
-    /// # Arguments
-    ///
-    /// * `msg` - Message content to send
-    ///
-    /// * `event_id` - ID of the event to reply to
-    ///
-    /// * `is_reply` - Whether the message is a reply on a thread
-    pub async fn send_thread_reply(
-        &self,
-        msg: Arc<RoomMessageEventContentWithoutRelation>,
-        event_id: String,
-        is_reply: bool,
-    ) -> Result<(), ClientError> {
-        let event_id = EventId::parse(event_id)?;
+        let reply = reply_params.to_reply()?;
         self.inner
-            .send_reply(
-                (*msg).clone(),
-                event_id,
-                EnforceThread::Threaded(if is_reply {
-                    ReplyWithinThread::Yes
-                } else {
-                    ReplyWithinThread::No
-                }),
-            )
+            .send_reply((*msg).clone(), reply.event_id, reply.enforce_thread)
```

Comment:
This could maybe use a single parameter of type `Reply` rather than separate parameters for the event ID and threading behaviour. The same applies to a few other methods down the call tree and would probably require moving `Reply` from `attachment.rs` into `repply.rs`.

Happy to make those changes if that seems reasonable.